src/olm/messages/message.rs

```diff
@@ -264,6 +389,49 @@ struct ProtoBufMessage {
     ciphertext: Vec<u8>,
 }
 
+#[cfg(feature = "interolm")]
+#[derive(PartialEq, Eq, ProstMessage)]
+struct InterolmProtoBufMessage {
+    #[prost(bytes, tag = "1")]
+    ratchet_key: Vec<u8>,
+    #[prost(uint32, tag = "2")]
+    counter: u32,
+    #[prost(uint32, tag = "3")]
+    previous_counter: u32,
+    #[prost(bytes, tag = "4")]
+    ciphertext: Vec<u8>,
+}
+
+#[cfg(feature = "interolm")]
+impl InterolmProtoBufMessage {
+    const RATCHET_TAG: &'static [u8; 1] = b"\x0A";
+    const INDEX_TAG: &'static [u8; 1] = b"\x10";
+    const PREVIOUS_INDEX_TAG: &'static [u8; 1] = b"\x18";
```

Comment:
Maybe `COUNTER_TAG` and `PREVIOUS_COUNTER_TAG` would match the property names better?

Reply:
We have a mix of the `counter` and `index` nomenclature in various places, including in Megolm (unrelated to this PR). We prefer the `index` terminology, as in the existing Olm implementation, so I've replaced the remaining references to `counter` with it.

The few remaining Megolm locations are best left to a separate PR.